vendor/bat-native-ledger/src/bat/ledger/internal/promotion/promotion.h

```diff
@@ -98,15 +122,35 @@ class Promotion {
       const std::string promotion_string,
       ledger::ResultCallback callback);
 
-  bool UnBlindTokens(
+  void ProcessSignedCredentials(
+      const ledger::Result result,
+      const std::string& promotion_id,
+      ledger::ResultCallback callback);
+
+  void OnProcessSignedCredentials(
       ledger::PromotionPtr promotion,
-      std::vector<std::string>* unblinded_encoded_tokens);
+      ledger::ResultCallback callback);
 
-  void FinishPromotion(
+  void SaveUnblindedTokens(
       ledger::PromotionPtr promotion,
       const std::vector<std::string>& unblinded_encoded_tokens,
       ledger::ResultCallback callback);
 
+  void FinishPromotion(
+      const ledger::Result result,
+      const std::string promotion_string,
+      ledger::ResultCallback callback);
+
+  void CheckForCorrupted(ledger::PromotionMap promotions);
```

Comment:
If it's really an STL map under the covers, should probably be a const reference if we're not mutating it (pointer if we are mutating it).

Reply:
done